src/admin/resolvers/queries/ApprovedItem.ts

```diff
@@ -76,3 +77,53 @@ export async function getApprovedItemByUrl(
 
   return await dbGetApprovedItemByUrl(context.db, args.url);
 }
+
+/**
+ * Retrieves a history of when and on what scheduled surface a corpus item
+ * was scheduled.
+ */
+export async function getScheduledSurfaceHistory(
+  parent,
+  args,
+  { db }
+): Promise<ScheduledItem[]> {
+  // Get the external ID of the Approved Item itself
+  const { externalId } = parent;
+
+  // Get the optional filters. Not specifying any means we will retrieve
+  // the entire list of all the occasions this approved item was scheduled
+  // on all surfaces, in descending order (most recently scheduled first).
+  //
+  // Limiting it to a single surface means we'll only retrieve data for
+  // one scheduled surface (useful for when you only need the one to filter out
+  // prospects on the Prospecting page).
+  //
+  // Limiting the number of results to one means only the most recent result
+  // will be returned.
+  const { scheduledSurfaceGuid, limit } = args;
+
+  // Check if the scheduled surface supplied is valid
+  const surface = ScheduledSurfaces.find((surface) => {
+    return surface.guid === scheduledSurfaceGuid;
+  });
+  if (!surface) {
```

Comment:
IMO you could skip this check and just rely on no results being returned for the `where` condition

Reply:
I could, but I would lose the ability to return an input error to the client, which is something I'd like to do to make the frontend side of things easier during development.